### PR TITLE
Adding policy to login and adding web signup endpoint

### PR DESCRIPTION
**Pull request by soareswallace**
## Proposed changes

TBA

## Further comments

TBA

## Checklist
- [x] Style check and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/incognia/api/IncogniaAPI.java [hunk 1]
```diff
@@ -252,6 +256,67 @@ public TransactionAssessment registerWebLogin(RegisterWebLoginRequest request)
...
+        new PostSignupRequestBody(
+            null,
+            request.getSessionToken(),
+            request.getAddress().getAddressLine(),
+            request.getAddress().getStructuredAddress(),
+            request.getAddress().getCoordinates(),
+            request.getExternalId(),
+            request.getPolicyId(),
+            request.getAccountId(),
+            request.getAdditionalLocations());
```
Comment (alangalvino) [inline src/main/java/com/incognia/api/IncogniaAPI.java:315]: Acho que não existe additional_locations para web

File: src/main/java/com/incognia/api/IncogniaAPI.java [hunk 2]
```diff
@@ -252,6 +256,67 @@ public TransactionAssessment registerWebLogin(RegisterWebLoginRequest request)
...
+    Asserts.assertNotNull(request, "register signup request");
+    Asserts.assertNotEmpty(request.getSessionToken(), "session token");
+    Asserts.assertNotNull(request.getAddress(), "address");
+    PostSignupRequestBody postSignupRequestBody =
+        new PostSignupRequestBody(
+            null,
+            request.getSessionToken(),
+            request.getAddress().getAddressLine(),
+            request.getAddress().getStructuredAddress(),
+            request.getAddress().getCoordinates(),
```
Comment (alangalvino) [inline src/main/java/com/incognia/api/IncogniaAPI.java:311]: Acho que não existe endereço para web

File: src/main/java/com/incognia/api/IncogniaAPI.java (hunk 1 above)
Comment (soareswallace) [reply src/main/java/com/incognia/api/IncogniaAPI.java:315]: Removi localmente ja

File: src/main/java/com/incognia/api/IncogniaAPI.java (hunk 2 above)
Comment (soareswallace) [reply src/main/java/com/incognia/api/IncogniaAPI.java:311]: Removi localmente ja

File: src/main/java/com/incognia/onboarding/RegisterWebSignupRequest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.incognia.common.Address;
+import java.util.List;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class RegisterWebSignupRequest {
+  String sessionToken;
+  Address address;
```
Comment (alangalvino) [inline src/main/java/com/incognia/onboarding/RegisterWebSignupRequest.java:13]: Não existe address em web

File: src/main/java/com/incognia/onboarding/RegisterWebSignupRequest.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  String externalId;
+  String policyId;
+  String accountId;
+  List<AdditionalLocation> additionalLocations;
```
Comment (alangalvino) [inline src/main/java/com/incognia/onboarding/RegisterWebSignupRequest.java:17]: Não existe additional locations em web

File: src/main/java/com/incognia/api/IncogniaAPI.java [hunk 5]
```diff
@@ -186,6 +188,7 @@ public TransactionAssessment registerLogin(RegisterLoginRequest request)
             .installationId(request.getInstallationId())
             .accountId(request.getAccountId())
             .externalId(request.getExternalId())
+            .policyId(request.getPolicyId())
```
Comment (racevedoo) [inline src/main/java/com/incognia/api/IncogniaAPI.java:191]: shouldn't we also add policy id to payment requests?

**Review by racevedoo — APPROVED** [review]

File: src/main/java/com/incognia/api/IncogniaAPI.java (hunk 5 above)
Comment (soareswallace) [reply src/main/java/com/incognia/api/IncogniaAPI.java:191]: Maybe we should. But I think it shouldn't be on this PR.

Comment (ebaf) [reply src/main/java/com/incognia/api/IncogniaAPI.java:191]: +1

**Pull request merged**